pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -224,49 +224,99 @@ func (metrics *vmiMetrics) updateBlock(blkStats []stats.DomainStatsBlock) {
 			continue
 		}
 
-		if block.RdReqsSet || block.WrReqsSet {
-			desc := metrics.newPrometheusDesc(
-				"kubevirt_vmi_storage_iops_total",
-				"I/O operation performed.",
-				[]string{"drive", "type"},
+		blkLabels := []string{"drive"}
+		blkLabelValues := []string{block.Name}
+
+		if block.Alias != "" {
+			blkLabelValues[0] = block.Alias
+		}
+
+		if block.RdReqsSet {
+			metrics.pushCustomMetric(
+				"kubevirt_vmi_storage_iops_read_total",
+				"I/O read operations",
+				prometheus.CounterValue,
+				float64(block.RdReqs),
+				blkLabels,
+				blkLabelValues,
 			)
+		}
 
-			if block.RdReqsSet {
-				metrics.pushPrometheusMetric(desc, prometheus.CounterValue, float64(block.RdReqs), []string{block.Name, "read"})
```

Comment:
Why is the `read` and `write` label removed? This makes it very hard to create sums and is an antipattern. :/

Reply:
For others: After the discussions about kubevirt metrics, this splitting is approved at the moment.

Reply:
I was a little bit disappointed that there was not more input on the metrics doc. Like background on why they chose that split and so on. But since no one else seems to take interest in this, I am not blocking, although I can't say that I am a fan of it.